### ISSUE TITLE
Unison fails on OSX if "-follow" and "-repeat watch" are set

### ISSUE THREAD

**Issue by yannickmodahgouez** [bcpierce00/unison#79]
I am using UNISON 2.50.0 (OCAML 4.03.0) compiled directly from the git repo.

If I sync a folder containing a symlink using the "-follow" parameter it works fine.
However, setting the "-repeat watch" parameter will result in the following crash : 

> Fatal error: File monitoring helper program not running

**Comment by alkuzad** [comment]
do you have unison-fsmonitor somewhere in PATH ?

**Comment by yannickmodahgouez** [comment]
yes I do. And "-repeat watch" works fine when no "-follow" parameter is set.

**Comment by alkuzad** [comment]
On Linux I can't reproduce with 2.48.3, we don't use OSX so often here in Europe. 


```
mkdir -p ~/test/a ~/test/b ~/test/c

unison ~/test/a ~/test/b -follow 'Path {.*testlink.*}' -repeat watch &

cd ~/test/a
ln -s ~/test/c teslink

UNISON 2.48.3 started propagating changes at 09:29:12.17 on 05 Jun 2017
[BGN] Copying testlink from /home/alkuzad/test/a to /home/alkuzad/test/b
[END] Copying testlink
UNISON 2.48.3 finished propagating changes at 09:29:12.18 on 05 Jun 2017
Saving synchronizer state
Synchronization complete at 09:29:12  (1 item transferred, 0 skipped, 0 failed)
Looking for changes
Reconciling changes
Nothing to do: replicas have not changed since last sync.
```

**Comment by lasdjfk** [comment]
I have the same problem on Mac OS Sierra 10.12.5 with unison 2.48.4. 

When syncing with repeat watch and the follow option it works fine until I create a symlink. Then I get the same fatal error as above. However the syncing continues to work fine it seems, but the symlink is not followed.

Could have something to do with how unison-fsmonitor handles symlinks?

**Comment by m-f-g** [comment]
Mac OS 10.12.6
unison 2.48.4
with option -repeat watch I got: 
Fatal error: File monitoring helper program not running

**Comment by sfedosimov** [comment]
@m-f-g I fixed the issue with following steps:
1. brew install unison
2. Using https://github.com/hnsl/unox:
 - brew tap eugenmayer/dockersync
 - brew install eugenmayer/dockersync/unox

**Comment by bcpierce00** [comment]
Any idea why this helped?

**Comment by seawatts** [comment]
I have tried https://github.com/bcpierce00/unison/issues/79#issuecomment-336738747 and it still did not work

**Comment by frantzmiccoli** [comment]
Comment by @sfedosimov https://github.com/bcpierce00/unison/issues/79#issuecomment-336738747 works here, I would say that this is likely installing a missing dependency somewhere.

**Comment by robertocaldas** [comment]
The unox trick worked for me, but only after doing `brew link unox`

**Comment by adielsa** [comment]
You might miss the unison-fsmonitor on your macOS side
brew install autozimu/homebrew-formulas/unison-fsmonitor

**Comment by vamega** [comment]
I seem to be running into this as well. I'm running 2.53.2 at the moment, will test wiht 2.53.3 tonight.
This only happens when I uncomment a `follow` directive. `repeat` on it's own works fine.

I'm using `autozimu/unison-fsmonitor` on a mac, and the remote end is Linux with the exact same version.
`unison-fsmonitor` exists on the PATH for both the local and remote machines.

Remote machine's PATH is set correctly even when executing in a non-interactive terminal.

```
❯ ssh remote-machine 'which unison-fsmonitor'
/home/madiath/.nix-profile/bin/unison-fsmonitor
```

**Comment by vamega** [comment]
I plan to at least get good debug logs and provide those.
I'm not sure how to run the fsmonitor with debug logs, any guidance there would be helpful.

I'm guessing that it's not as simple as running Unsion with debug logs?

**Comment by gdt** [comment]
The autozimu fsmonitor is not from the unison project.  I suggest asking them for how to enable debugging.   Since this is an interaction issue between two programs from different sources, it seems really necessary to have debugging on both sides.

If you need help debugging, please feel free to ask on the mailnglist, but that's out of scope in the tracker.
